source/SpinalHDL/Data types/AFix.rst

```diff
@@ -0,0 +1,205 @@
+
+.. _AFix:
+
+AFix
+====
+
+Description
+^^^^^^^^^^^
+
+Auto-ranging Fixed-Point, ``AFix``, is a fixed-point class which tracks the representable range of values while preforming fixed-point operations.
+
+
+Declaration
+^^^^^^^^^^^
+
+AFix can be created using bit sizes or exponents:
+
+.. code-block:: scala
+
+  U(12 bits)        // U12.0
+  UQ(8 bits, 4 bits) // U8.4
+  U(8 exp, 12 bits) // U8.4
+  U(8 exp, -4 exp) // U8.4
+  U(8 exp, 4 exp)  // U8.-4
+
+  S(12 bits)        // S11 + sign
+  SQ(8 bits, 4 bits) // S8.4 + sign
+  S(8 exp, 12 bits) // S8.3 + sign
+  S(8 exp, -4 exp) // S8.4 + sign
+
+
+These will have representable ranges for all bits.
+
+For example:
+
+``U(12 bits)`` will have a range of 0 to 4095.
+
+``SQ(8 bits, 4 bits)`` will have a range of -4096 (-256) to 4095 (255.9375)
+
+``U(8 exp, 4 exp)`` will have a range of 0 to 256
+
+
+Custom range ``AFix`` values can be created be directly instantiating the class.
+
+.. code-block:: scala
+
+  AFix(0, 4096, 0 exp)
+  AFix(-256, 256, -2 exp)
```

Comment:
Without comments it can be confusing, as first arg isn't the min value, but the the number of time the resolution.